src/cijoe/core/command.py

```diff
@@ -196,7 +203,17 @@ def _run(self, cmd, cwd, env, transport):
 
         return err, state
 
-    def run(self, cmd, cwd=None, env={}):
+    def _get_transport(self, endpoint=None):
+        if not endpoint:
+            endpoint = next(iter(self.transports))
+        if endpoint not in self.transports:
```

Comment:
Maybe this should be an `else if` since if the first is condition is true, this will always be true as well.
I initially misread the code, perhaps this change could make it a bit clearer.

Reply:
I know that if line 207 evaluates to true, then 209 will always evaluate to false (as the code is now), but because we change the value of `endpoint` (now renamed to `transport_name`) in line 208, I prefer to use another `if` over an `elif`. If we mess with line 208 in the future, line 209 is there to ensure that it doesn't break.
It's a constant-time lookup, so performance wise, it makes no difference, so I believe it's just a matter of preference.